src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -250,17 +255,19 @@ static boolean passesBasicCvrSourceValidation(CvrSource source) {
         }
       }
 
-      if (isNullOrBlank(source.getContestId()) &&
-          (provider == Provider.DOMINION || provider == Provider.HART
-              || provider == Provider.CLEAR_BALLOT)) {
+      boolean providerRequiresContestId = provider == Provider.DOMINION ||
+          provider == Provider.HART ||
+          provider == Provider.CLEAR_BALLOT ||
+          provider == Provider.CDF;
```

Comment:
Apologies if I missed the earlier discussion on this point: why are we adding this requirement? Is it necessary for both JSON and XML? Why didn't we need to have it before?

Reply:
It should have been there all along.  CDF supports multiple contests, therefore we need to specify which contest to tabulate.  Our CDF json examples created by George only contained one contest each, so it may have seemed appropriate at the time to just assume that the one contest is 'the one to tabulate' but that is not the case.

Reply:
Please update the `config_file_documentation.txt` section on `contestId` to reflect this. It currently only mentions Hart and Dominion. It'd also be great if you could make a quick pass on that file with the new validation stuff you're adding here to make sure there aren't any other notes that should be added.